package dnb.atm.dao.impl;

import java.net.ConnectException;

import org.junit.Test;
import org.junit.runner.RunWith;
import org.mockito.InjectMocks;
import org.mockito.Mock;
import org.mockito.Mockito;
import org.mockito.runners.MockitoJUnitRunner;
import org.springframework.beans.factory.annotation.Autowired;
import org.springframework.test.context.ContextConfiguration;
import dnb.atm.dao.LoginDao;
import dnb.atm.model.CustomerModel;
import junit.framework.Assert;

/***
 * 
 * @author phuctd.lt
 *
 */

@RunWith(MockitoJUnitRunner.class)
@ContextConfiguration(locations = { "file:src/test/resources/unittest-config-context.xml" })
public class LoginDaoImplTest {
	@InjectMocks
	private LoginDaoImpl loginDaoImpl = new LoginDaoImpl();
	@Mock
	private LoginDao loginDao;
	@Test
	public void loginDaoTest() throws ConnectException{
		CustomerModel obj = new CustomerModel();
		obj.setAccountID("0107744763");
		String user = "0107744763";
		String pass = "123456" ;
		Assert.assertEquals(obj, loginDao.login(user, pass));
		
	}

}
